src/test/java/core/basesyntax/service/impl/TransactionHandlerImplTest.java

```diff
@@ -0,0 +1,85 @@
+import core.basesyntax.strategy.operations.ReturnOperationHandler;
+import core.basesyntax.strategy.operations.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransactionHandlerImplTest {
+    private static final String TITLE_FOR_BANANAS = "banana";
+    private static final String TITLE_FOR_APPLES = "apple";
+    private static final Integer QUANTITY_FOR_BANANAS = 152;
+    private static final Integer QUANTITY_FOR_APPLES = 90;
+    private static final List<FruitTransaction> EMPTY_FRUIT_TRANSACTIONS = Collections.emptyList();
+    private static final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap
+            = new HashMap<>();
+    private static List<FruitTransaction> validFruitTransactions;
+    private static TransactionHandler transactionHandler;
+    private static OperationStrategy operationStrategy;
+
+    @BeforeClass
+    public static void beforeClass() {
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperationHandler());
+        operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+        transactionHandler = new TransactionHandlerImpl(operationStrategy);
+        validFruitTransactions = new ArrayList<>();
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                TITLE_FOR_BANANAS, 20));
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                TITLE_FOR_APPLES, 100));
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                TITLE_FOR_BANANAS, 100));
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                TITLE_FOR_BANANAS, 13));
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.RETURN,
+                TITLE_FOR_APPLES, 10));
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                TITLE_FOR_APPLES, 20));
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                TITLE_FOR_BANANAS, 5));
+        validFruitTransactions.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                TITLE_FOR_BANANAS, 50));
+    }
+
+    @Test (expected = RuntimeException.class)
+      public void processData_emptyListOfData_notOk() {

```

Comment:
don`t fixed

Reply:
Why do I need to fix this test? It works correctly. I check here the work of the method for a passed empty input list. If the input list is empty, you need to throw Runtime Exception.